lib/Controller/Text2ImageApiController.php

```diff
@@ -119,21 +114,31 @@ public function getImage(string $imageGenId, int $fileNameId): DataDisplayRespon
 	}
 
 	/**
+	 * Get image generation information
+	 *
 	 * @param string $imageGenId
-	 * @return DataResponse
+	 * @return DataResponse<Http::STATUS_OK, AssistantImageGenInfo, array{}>|DataResponse<Http::STATUS_BAD_REQUEST|Http::STATUS_NOT_FOUND|Http::STATUS_INTERNAL_SERVER_ERROR, array{error: string}, array{}>
+	 *
+	 * 200: Returns the requested data
+	 * 400: The image generation does not exist
+	 * 404: The image generation has been deleted
+	 * 500: Other error
 	 */
 	#[NoAdminRequired]
 	#[NoCSRFRequired]
 	#[PublicPage]
 	#[BruteForceProtection(action: 'imageGenId')]
+	#[OpenAPI(scope: OpenAPI::SCOPE_DEFAULT, tags: ['image_generation'])]
 	public function getGenerationInfo(string $imageGenId): DataResponse {
 		try {
 			$result = $this->text2ImageHelperService->getGenerationInfo($imageGenId, $this->userId, true);
 		} catch (Exception $e) {
-			$response = new DataResponse(['error' => $e->getMessage()], (int) $e->getCode());
-			if ($e->getCode() === Http::STATUS_BAD_REQUEST || $e->getCode() === Http::STATUS_UNAUTHORIZED) {
+			/** @var Http::STATUS_BAD_REQUEST|Http::STATUS_NOT_FOUND|Http::STATUS_INTERNAL_SERVER_ERROR $exceptionCode */
+			$exceptionCode = (int) $e->getCode();
```

Comment:
Wondering if there is a case where $exceptionCode is not a valid http status code...

Reply:
As we are the ones throwing the exception, we know which codes we can get here. I don't like how it's implemented, I just made the necessary changes for the psalm check to be green. This would definitely need a bit of refactoring/cleaning.